### DISCUSSION TITLE
sharding and replication_factor for a collections

### DISCUSSION THREAD

**Discussion by mm3906078** [qdrant/qdrant#4340] [Q&A] [answered]
Hi friends, I have a question about sharding. 
The document said using more than one shardes for each nodes is better. For example, if I have five nodes and I set sharding to 10, and all my data have 10G, how will the data expand in the cluster?

If each server has ten shards and all of my data is in them, why can I lose two nodes, not 4?

**Comment by mm3906078** [comment]
I should mention. I am confused because, in the dashboard, you can see all ten shardes marked as local and present on one node.

![image](https://github.com/qdrant/qdrant/assets/49244782/c372c5f7-a702-481e-8df5-5ada804d29e9)


 node: 5  
 shard_number: 10  
 replication_factor: 5

**Reply by timvisee** [reply, marked as answer]
Each single point is in all replicas of a single shard.

If you have a replication factor of 5, it means you have 5 copies of all your data across your nodes. 5 replications also mean that 4 nodes may fail without taking the whole cluster down.

Above you mention you have 5 nodes. In such case I'd recommend one of the two following configurations to minimize the amount of data on your cluster:

- shard_number: 10, replication_factor: 2 - allows at most 1 node to fail, has 2 copies of all points across the cluster
- shard_number: 10, replication_factor: 3 - allows at most 2 nodes to fail, has 3 copies of all points across the cluster

**Reply by mm3906078** [reply]
Thank you for your answer, @timvisee.

I understand your explanation, but I have another question. If I increase the replication factor to match the number of nodes (in this example, 5), I realize that I will be storing redundant data. Considering the RAFT protocol, I know I can lose up to 2 nodes in my cluster while still maintaining functionality, not more. However, will this increase the speed of my cluster, given that I am not using remote writing for my requests?

By the way, is it better to have 10 nodes with 30GB RAM each or 5 nodes with 60GB RAM each? I think 5 nodes are better because the network is always slower and RAM is much faster.

Sorry for taking up your time. I hope you are having a good day, my friend.

**Reply by timvisee** [reply]
Yes, matching the replication factor to have all data on all nodes will make it quicker. Though, the difference may be insignificant if the nodes are close to each other in terms of network time. It's a balance between performance and redundant data/memory/storage use.

Having more RAM on a single machine is probably better than spreading it out over more nodes. Though there is more to it, such as having a properly sized CPU to allow adequate indexing.

I'd highly recommend to properly try a few scenarios. That way you can validate your requirements and see what works best for you. There is a bunch of other parameters that greatly affect the performance of different aspects too, for example, `on_disk` to put all your data instead of in memory.

We have a tool called [`bfb`](https://github.com/qdrant/bfb) which might help you with that. It generates random data and searches over it.

**Reply by mm3906078** [reply]
Oh, Thank you for introducing me to bfb.
